centml/cli/cluster.py

```diff
@@ -20,10 +21,46 @@ def convert(self, value, param, ctx):
             return None  # to avoid warning from lint for inconsistent return statements
 
 
-hw_to_id_map = {"small": 1000, "medium": 1001, "large": 1002}
-id_to_hw_map = {v: k for k, v in hw_to_id_map.items()}
+# Cache hardware instances
+_hw_id_map_cache = None
+# Declare global variables with type annotations
+hw_to_id_map: Dict[str, int] = {}
+id_to_hw_map: Dict[str, str] = {}
 
 
+def fetch_hardware_mappings():
+    global _hw_id_map_cache
+    if _hw_id_map_cache is None:
+        _hw_id_map_cache = api.get_hardware_instances()
+    return _hw_id_map_cache
+
+
+def get_hw_to_id_map():
+    global hw_to_id_map, id_to_hw_map
+
+    response = fetch_hardware_mappings()
+
+    # Handle empty response
+    if not response:
+        click.echo("Failed to fetch hardware instances. Please try again.")
```

Comment:
it also doesn't make sense to show this error message and return an outdated instance list.

Reply:
but if the api fails for hardware instances then user will not be able to create the deployment so in that case what message do you want to show to the user as a fallback message?